### ISSUE TITLE
RevEng: Should we always specify .ForeignKey(...) when configuring relatioships

### ISSUE THREAD

**Issue by rowanmiller** [aspnet/EntityFramework#1937]
Currently we don't specify the FK properties (which is OK because we would generate ones that match convention). However, I think it would be clearer if we just completely specified the relationship.

```
entity.Reference(d => d.Category).InverseCollection(p => p.Product);
```

Would become

```
entity.Reference(d => d.Category).InverseCollection(p => p.Product).ForeignKey(d => d.CategoryId);
```

@divega @anpete @ajcvickers @bricelam thoughts?

**Comment by bricelam** [comment]
It would make it easier to rename the navigation properties on the dependent entity type...

**Comment by lajones** [comment]
Fixed by PR #1968
